src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs

```diff
@@ -15,140 +17,16 @@ internal static unsafe class PixelFormatTranscoder
         PixelFormat destFormat,
         AlphaFormat destAlphaFormat)
     {
-        var reader = GetReader(srcFormat);
-        var writer = GetWriter(destFormat);
+        var pixelCount = srcSize.Width * srcSize.Height;
+        var bufferSize = pixelCount * Marshal.SizeOf<Rgba8888Pixel>();
+        var blob = new UnmanagedBlob(bufferSize);
```

Comment:
Use `using var` instead of manual Dispose

Reply:
UnmanagedBlob isn't IDisposable

Reply:
Should probably make it to implement IDisposable, it has Dispose method and the entire finalizer pattern.